### PR TITLE
Added description discount  for each summary(cart, order)

### PR DESCRIPTION
**Pull request by MaxSouza**
### Description (*)

Added description discount  for each summary(cart, order)

### Fixed Issues (if relevant)

1. magento/magento2#3594: Discount Description does not get displayed on cart page

### Manual testing scenarios (*)

1. Create a coupon with description
2. Add product to cart and apply coupon
3. Check if discount description is displayed on summary

### Questions or comments

image
image

image

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by kalpmehta** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta, here is your [Magento instance](https://i-24595-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-24595-2-3-develop.instances.magento-community.engineering/admin](https://i-24595-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by kalpmehta** [comment]
@MaxSouza I don't think this is an issue. Showing shopping cart rule description may not work for everybody. Some merchants are using it for internal purpose, some might have quite bigger description text which cannot fit in the cart totals box. This is more like a workaround for the merchants who want to use description field to display some kind of message near the discount label.

**Comment by MaxSouza** [comment]
@kalpmehta Can you comment on issue, please?

**Comment by kalpmehta** [comment]
Sure @MaxSouza. Thank you for your contribution. I will comment on the issue and will also wait for some other reviewer if they think otherwise.

**Comment by kalpmehta** [comment]
@MaxSouza Can you please fix the space issues and add the tests?

File: app/code/Magento/SalesRule/Model/RulesApplier.php [hunk 1]
```diff
@@ -3,15 +3,19 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+
```
Comment (kalpmehta) [inline app/code/Magento/SalesRule/Model/RulesApplier.php]: please remove this empty line

File: app/code/Magento/SalesRule/Model/RulesApplier.php [hunk 2]
```diff
@@ -40,9 +44,9 @@ class RulesApplier
 
     /**
      * @param \Magento\SalesRule\Model\Rule\Action\Discount\CalculatorFactory $calculatorFactory
-     * @param \Magento\Framework\Event\ManagerInterface $eventManager
-     * @param \Magento\SalesRule\Model\Utility $utility
-     * @param ChildrenValidationLocator|null $childrenValidationLocator
+     * @param \Magento\Framework\Event\ManagerInterface                       $eventManager
```
Comment (kalpmehta) [inline app/code/Magento/SalesRule/Model/RulesApplier.php]: Please remove the tabs and extra spaces, follow Magento's formatting

File: app/code/Magento/SalesRule/Model/RulesApplier.php [hunk 3]
```diff
@@ -109,8 +113,8 @@ public function applyRules($item, $rules, $skipValidation, $couponCode)
     /**
      * Add rule discount description label to address object
      *
-     * @param Address $address
-     * @param \Magento\SalesRule\Model\Rule $rule
+     * @param  Address $address
+     * @param  Rule    $rule
```
Comment (kalpmehta) [inline app/code/Magento/SalesRule/Model/RulesApplier.php]: please remove extra spaces

**Review by kalpmehta — CHANGES_REQUESTED** [review]

**Comment by MaxSouza** [comment]
@kalpmehta I'm working on tests

**Comment by kalpmehta** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta, here is your [new Magento instance](https://pr-24595.instances.magento-community.engineering).
Admin access: [https://pr-24595.instances.magento-community.engineering/admin](https://pr-24595.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Review by kalpmehta — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta, thank you for the review.
[ENGCOM-5926](https://jira.corp.magento.com/browse/ENGCOM-5926) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Comment by aman3103** [comment]
@magento-engcom-team I don't think this is a good change to show the coupon description in admin also as customer care team then don't have any information about which code is applied as in coupon code field also they don't have way to filter the code using description.

**Pull request merged**